### PR TITLE
Tb 2029 add whitelisting

### PR DESCRIPTION
**Pull request by joecks**
Implement iOS handling for whitelisting

### CODE REVIEW

File: packages/webview_flutter/ios/Classes/ContentBlocker.swift [hunk 1]
```diff
@@ -147,6 +193,25 @@ class WKContentRuleBlocker: ContentBlockerApi {
             )
     }
 
+    fileprivate func configureWebview(key: String, webview: FLTWKWebView) {
+        var shouldAdd = true
+        if let host = webview.url?.host {
```
Comment (peterstojanowski) [inline packages/webview_flutter/ios/Classes/ContentBlocker.swift:198]: These 3 checks can be done in one `if` statement.

File: packages/webview_flutter/ios/Classes/ContentBlocker.swift [hunk 2]
```diff
@@ -147,6 +193,25 @@ class WKContentRuleBlocker: ContentBlockerApi {
...
+            if let entry = whiteListing[host] {
+                if (entry[key] == ACTIVE) {
+                    print("Host \(host) has active \(key) feature.")
+                    shouldAdd = false
+                }
+            }
+        }
+        if (shouldAdd) {
```
Comment (peterstojanowski) [inline packages/webview_flutter/ios/Classes/ContentBlocker.swift:206]: Don't use brackets in Swift for `if` statements.

File: packages/webview_flutter/ios/Classes/ContentBlocker.swift [hunk 3]
```diff
@@ -37,21 +44,61 @@ protocol ContentBlockerApi {
...
+                dict[entry.key] = value
+                return dict
+            }
+        }
+        whiteListing = tempDict
+    }
+
+    @objc public func onUrlChanged(webview: FLTWKWebView) {
+        for list in compiledLists.keys {
+            self.configureWebview(key: list, webview: webview)
```
Comment (peterstojanowski) [inline packages/webview_flutter/ios/Classes/ContentBlocker.swift:98]: `self` id not needed here.

**Review by peterstojanowski — CHANGES_REQUESTED** [review]

File: packages/webview_flutter/ios/Classes/ContentBlocker.swift [hunk 4]
```diff
@@ -147,6 +194,27 @@ class WKContentRuleBlocker: ContentBlockerApi {
             )
     }
 
+    fileprivate func configureWebview(key: String, webview: FLTWKWebView) {
+        var shouldAdd = shouldAddByDefault(key: key)
+        let host = webview.url?.host
+        if host != nil && whiteListing[host!]?[key] == ACTIVE {
```
Comment (peterstojanowski) [inline packages/webview_flutter/ios/Classes/ContentBlocker.swift:200]: `if let host = webview.url?.host, whiteListing[host]?[key] == ACTIVE {}`

File: packages/webview_flutter/ios/Classes/ContentBlocker.swift [hunk 5]
```diff
@@ -147,6 +194,27 @@ class WKContentRuleBlocker: ContentBlockerApi {
...
+            print("Loaded \(key) content rule set")
+        } else {
+            removeContentBlocker(key: key, webview: webview)
+            print("Removed \(key) content rule set")
+        }
+    }
+
+    fileprivate func shouldAddByDefault(key: String) -> Bool {
+        let entry = whiteListing[DEFAULT_WHITELISTING_KEY]
+        return entry == nil || entry![key] != ACTIVE
```
Comment (peterstojanowski) [inline packages/webview_flutter/ios/Classes/ContentBlocker.swift:215]: guard let entry = whiteListing[DEFAULT_WHITELISTING_KEY] else { return true }
return entry[key] != ACTIVE

File: packages/webview_flutter/ios/Classes/ContentBlocker.swift (hunk 4 above)
Comment (joecks) [reply packages/webview_flutter/ios/Classes/ContentBlocker.swift:200]: wtf, who would guess you can seperate those statements by a comma

**Review by peterstojanowski — APPROVED** [review]

**Pull request merged**